app/scripts/modules/core/src/pipeline/config/stages/executionWindows/ExecutionWindows.tsx

```diff
@@ -0,0 +1,409 @@
+                    restrictExecutionDuringTimeWindow: !props.restrictExecutionDuringTimeWindow,
+                  })
+                }
+              />
+            </div>
+          )}
+        </div>
+      </div>
+      {props.restrictExecutionDuringTimeWindow && (
+        <div>
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <h5>
+                <strong>Days of the Week</strong>
+                <small>(No days selected implies execution on any day if triggered)</small>
+              </h5>
+              <div className="execution-window-days-of-week">
+                <ExecutionWindowDayPicker days={days} onChange={windowDaysUpdated} />
+              </div>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <p>
+                <strong>Time of Day</strong>
+              </p>
+            </div>
+          </div>
+          <div className="row">
+            <div className="col-md-10 col-md-offset-1">
+              <div className="execution-window-graph">
+                <div className="execution-day">
+                  {timelineWindows.map((w, i) => {
+                    return (
+                      <Tooltip key={i} value={moment(w.start).format('HH:mm') + ' - ' + moment(w.end).format('HH:mm')}>
+                        <div className="execution-window" style={w.style} />
+                      </Tooltip>
+                    );
+                  })}
+                </div>
+                <div className="divider-label-fill" />
+                <div className="dividers">
+                  {dividers.map((d, i) => {
+                    return (
+                      <div className="divider" key={i} style={{ left: d.left }}>
+                        <div className="divider-filler" />
+                        <div className="divider-label">{d.label}</div>
+                      </div>
+                    );
+                  })}
+                </div>
+              </div>
+            </div>
+          </div>
+          <div className="form-group">
+            <p className="col-md-9 col-md-offset-1 form-control-static">
+              This stage will only run within the following windows (all times in <SystemTimezone />
+              ):
+            </p>
+          </div>
+          {whitelist.map((w: IWindow, i: number) => {
+            return (
+              <div className="window-entry" key={i}>
+                <div className="form-group">
+                  <div className="col-md-6 col-md-offset-1">
+                    <span className="start-end-divider">From</span>
+                    <ReactSelectInput
+                      clearable={false}
+                      inputClassName={
+                        'visible-xs-inline-block visible-sm-inline-block visible-md-inline-block visible-lg-inline-block vertical-align-select'
```

Comment:
This class string gets repeated a few times below, and it's pretty long - it's probably worth moving it up to a variable and referencing that.

Reply:
👍